### DISCUSSION TITLE
CBC PriorityIn

### DISCUSSION THREAD

**Discussion by kartinator89** [coin-or/Cbc#350] [Q&A] [answered]
I'm using the CBC executable as a standalone black-box solver. I pass LP files to the CBC executable and write out the solution. It looks like there's a way to specify branching priorities from an input file using "priorityIn". The following is from the help manual in the executable.

> prio(rityIn) : Import priorities etc from file This will read a file with priorities from the given file name. It will use the default directory given by 'directory'. A name of '$' will use the previous value for the name. This is initialized to '', i.e. it must be set. This can not read from compressed files. File is in csv format with allowed headings - name, number, priority, direction, up, down, solution. Exactly one of name and number must be given.

Is there more detailed documentation or a sample input file explaining how to use this and how "name, number, priority, direction, up, down, solution" have to be specified?

**Comment by jjhforrest** [comment, marked as answer]
As a simple example for miplib/p0033 - if below is p0033.csv

name,priority
C157 , 100
C158 , 100
C159 , 100
C160 , 100
then after priorityin those would have priority of 100 and the rest 
would have their default priority of 1000.  The highest priority is 1 so 
these would be branched on before others.

You can go further and give up and down pseudo-costs or direction to 
branch on first - but I would not try to be too clever at first.  It is 
probably better to use mipstart than use the solution field.

John Forrest
